File: examples/chat-api/chat-completion.js
const apiKey = process.env.IZTRO_API_KEY;

if (!apiKey) {
  throw new Error('Set IZTRO_API_KEY before running this example.');
}

const response = await fetch('https://chat-api.iztro.com/v2/chat/completions', {
  method: 'POST',
  headers: {
    Authorization: `Bearer ${apiKey}`,
    'Content-Type': 'application/json',
  },
  body: JSON.stringify({
    model: 'ziwei-agent',
    messages: [
      {
        role: 'user',
        content:
          '分析我的 2026 年事业趋势。生日是 1995-02-23，出生时辰 17 点，性别女。',
      },
    ],
    enable_iztro_call: true,
  }),
});

if (!response.ok) {
  throw new Error(`Request failed: ${response.status} ${await response.text()}`);
}

console.log(JSON.stringify(await response.json(), null, 2));

File: examples/chat-api/chat_completion.py
import json
import os

import requests


api_key = os.environ.get("IZTRO_API_KEY")
if not api_key:
    raise RuntimeError("Set IZTRO_API_KEY before running this example.")

response = requests.post(
    "https://api.iztro.com/v1/chat/completions",
    headers={
        "Authorization": f"Bearer {api_key}",
        "Content-Type": "application/json",
    },
    json={
        "model": "ziwei-agent",
        "messages": [
            {
                "role": "user",
                "content": "分析我的 2026 年事业趋势。生日是 1995-02-23，出生时辰 17 点，性别女。",
            }
        ],
        "enable_iztro_call": True,
    },
    timeout=60,
)

response.raise_for_status()
print(json.dumps(response.json(), indent=2, ensure_ascii=False))
